draft-ietf-tls-esni.md

```diff
@@ -542,22 +545,25 @@ These requirements prevent an attacker from performing a packet amplification
 attack, by crafting a ClientHelloOuter which decompresses to a much larger
 ClientHelloInner. This is discussed further in {{decompression-amp}}.
 
-Implementations SHOULD bound the time to compute a ClientHelloInner
-proportionally to the ClientHelloOuter size. If the cost is disproportionately
-large, a malicious client could exploit this in a denial of service attack.
+Implementations SHOULD construct the ClientHelloInner in linear
+time. Quadratic time implementations (such as may happen via naive
+copying) create a denial of service risk.
 {{linear-outer-extensions}} describes a linear-time procedure that may be used
 for this purpose.
 
 ## Authenticating the ClientHelloOuter {#authenticating-outer}
 
-To prevent a network attacker from modifying the reconstructed ClientHelloInner
-(see {{flow-clienthello-malleability}}), ECH authenticates ClientHelloOuter by
-passing ClientHelloOuterAAD as the associated data for HPKE sealing and opening
-operations. The ClientHelloOuterAAD is a serialized ClientHello structure,
-defined in {{Section 4.1.2 of RFC8446}}, which matches the ClientHelloOuter
-except the `payload` field of the "encrypted_client_hello" is replaced with a
-byte string of the same length but whose contents are zeros. This value does
-not include the four-byte header from the Handshake structure.
+To prevent a network attacker from modifying the reconstructed
+ClientHelloInner by modifying extensions in ClientHelloOuter that are
+referenced in ClientHelloInner (see
```

Comment:
This text and the original text are right but a bit confusing. I am going to generalize them in a subsequent PR to make them a bit easier to understand at this point in the document.